google/cloud/firestore_v1/document.py

```diff
@@ -402,16 +403,26 @@ def get(
             create_time = None
             update_time = None
         else:
-            data = _helpers.decode_dict(document_pb.fields, self._client)
-            exists = True
-            create_time = document_pb.create_time
-            update_time = document_pb.update_time
+            batched_response: BatchGetDocumentsResponse = list(batch_response_pb)[0]
+            read_time = batched_response.read_time
+            document: Document = batched_response.found
+            # TODO: Verify this check is correct.
+            if self.id not in batched_response.missing:
+                data = _helpers.decode_dict(document.fields, self._client)
+                exists = True
+                create_time = document.create_time
+                update_time = document.update_time
+            else:
+                data = None
+                exists = False
+                create_time = None
+                update_time = None
```

Comment:
does it make sense to gather these inits? I see we set create, update time to none twice for instance

Reply:
I was wondering about that. Will change.